runelite-client/src/main/java/net/runelite/client/ui/CustomNotification.java

```diff
@@ -0,0 +1,160 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.ui;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.Rectangle;
+import java.awt.TrayIcon;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import javax.imageio.ImageIO;
+import javax.swing.GroupLayout;
+import javax.swing.ImageIcon;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.Timer;
+import lombok.extern.slf4j.Slf4j;
+import org.pushingpixels.substance.internal.SubstanceSynapse;
+
+@Slf4j
+class CustomNotification
+{
+	private static final int width = 360;
+	private static final int height = 100;
+	private static final int padding = 50;
+
+	private static BufferedImage runeliteIcon;
+
+	static void sendCustomNotification(
+		final String title,
+		final String message,
+		final TrayIcon.MessageType type,
+		final String subtitle,
+		final JFrame runeliteWindow)
+	{
+		Rectangle display = runeliteWindow.getGraphicsConfiguration().getBounds();
+
+		JFrame window = new JFrame();
+		window.setUndecorated(true);
+		window.setIconImage(ClientUI.ICON);
+		window.setFocusableWindowState(false);
+		window.setBounds(
+			display.x + display.width - width - padding,
+			display.y + display.height - height - padding,
+			width,
+			height
+		);
+		((JPanel) window.getContentPane()).putClientProperty(SubstanceSynapse.COLORIZATION_FACTOR, 1.0);
+
+		float[] hsbs = new float[3];
+		Color stock = window.getContentPane().getBackground();
+		Color.RGBtoHSB(stock.getRed(), stock.getGreen(), stock.getBlue(), hsbs);
+		final long start = System.currentTimeMillis();
+		Timer anim = new Timer(20, e ->
+		{
+			double seconds = (System.currentTimeMillis() - start) / 1000.d;
+			if (seconds > 2)
+			{
+				((Timer) e.getSource()).stop();
+				window.getContentPane().setBackground(stock);
+				return;
+			}
+			float lum = (((float) Math.sin(seconds / 2.d * 3.14d)) / (1 / (.5f - hsbs[2]))) + hsbs[2];
+			window.getContentPane().setBackground(new Color(Color.HSBtoRGB(hsbs[0], hsbs[1], lum)));
+		});
+		anim.setRepeats(true);
+		anim.start();
+
+		Timer closeTimer = new Timer(5000, e ->
+		{
+			window.setVisible(false);
+			window.dispose();
+			anim.stop();
+		});
+		closeTimer.start();
+
+		window.addMouseListener(new MouseAdapter()
+		{
+			@Override
+			public void mousePressed(MouseEvent e)
+			{
+				window.setVisible(false);
+				window.dispose();
+				anim.stop();
+				closeTimer.stop();
+			}
+		});
+
+		if (runeliteIcon == null)
```

Comment:
To not have everything in this method, initializing this icon in static constructor might make sense

Reply:
I lazy loaded it as it won't be used in many cases

Reply:
It will be "lazy" loaded even when created in static constructor. If no field in CustomNotification will never be accessed, then static constructor will never be called too.